### ISSUE TITLE
Pylint Reporting Incorrect Line Numbers With @clean File

### ISSUE THREAD

**Issue by tbpassin** [leo-editor/leo-editor#2862]
Of course, clicking on the incorrect line number takes the cursor to the wrong line.

Changing the same file to @file, the correct line numbers are reported.

**Comment by edreamleo** [comment]
@tbpassin Please send me both the external file and the .leo file that demonstrates the problem.

**Comment by tbpassin** [comment]
It turns out that the same problem occurs for the pyflakes check-on-save.  This suggests to me that there are some extraneous lines in the temporary file used for checking an @clean file.

Here's a minimal outline that shows the problem on my machine.  It includes a README node that explains the issue and  how to demonstrate it.

[at-clean-pylint-issue.zip](https://github.com/leo-editor/leo-editor/files/9583733/at-clean-pylint-issue.zip)

**Comment by edreamleo** [comment]
@tbpassin The example you sent me work for me.  The clickable links go to the expected lines.

**Comment by tbpassin** [comment]
Hmm, how could it be different? Puzzle, puzzle!

```
Leo 6.7.0-devel, devel branch, build 88e8a60a8e
2022-09-14 06:46:32 -0500
Python 3.10.4, PyQt version 6.3.0
Windows 10 AMD64 (build 10.0.19043) SP0
```

**Comment by edreamleo** [comment]
@tbpassin Please do as I asked, that is, send me the complete external file and a .leo file containing the corresponding `@clean` node.

**Comment by tbpassin** [comment]
Looking at one of the @clean external files outside of Leo, it actually still has sentinel lines in it.  I'll have to create a new external file and see if that's still the case or not.

**Comment by tbpassin** [comment]
Yes, that's what is happening.  I also looked at the little test outline I posted the other day.  Each @clean file I saved included after saving two sentinels at the start:
```
#@+leo-ver=5-thin
#@+node:tom.20220916073658.1: * @clean example.py
```
There is also the matching sentinel at the end, but that won't mess up the line count:

```
#@-leo
```

When Leo reads in the @clean file, it leaves those sentinels in place - but not displayed as comment lines in the body - or possibly re-creates them when writing the file.

I deleted the sentinel lines in an external text editor and saved, reloaded the file in Leo from the drive, and then saved it again.  The sentinel lines re-appeared.

I've seen this behavior in at least several recent versions of Leo, but I have no idea when it started.  I'm also drawing a blank on why it's happening to me but not to you.  Unless ... I always add a character, usually a space, to make sure Leo actually saves the external file (when I'm doing tests like this).  Could it be that Leo didn't actually save the external file for you because it had no changes to save?

**Comment by tbpassin** [comment]
I looked at an older @clean file I've had in my Workbook.leo outline and haven't looked at in ages. It didn't have the sentinels.  I added a space and saved, and now the file contains the sentinels.

```
Leo 6.7.0-devel, devel branch, build 88e8a60a8e
2022-09-14 06:46:32 -0500
Python 3.10.4, PyQt version 6.3.0
Windows 10 AMD64 (build 10.0.19043) SP0
```

**Comment by edreamleo** [comment]
@tbpassin Iirc, for `@clean`, Leo stores the entire outline in the .leo file, so in strange situations as you describe it may be tricky to get to a reasonable state.

I suggest the following:

- Change `@clean` to `@file`.
- Edit the outline (in Leo!), to remove unwanted sentinels.
- Save the outline.
- Reload the .leo file.
- Make sure the file is as you want.
- Change `@file` to `@clean`
- Save the outline.
- Look at the external file in an external editor.

Repeat as necessary :-)

I'm going to close this issue. Apparently Leo's line-numbering code is not the culprit.

**Comment by tbpassin** [comment]
I guess I will go along with closing it.  I'm a bit flummoxed, because I got this problem several times on several different outlines.  Yet tonight, loading some of those misbehaving files, I noticed that I could see the sentinels (in Leo) even though the the file was marked @clean.

When as you suggested I edited the sentinels out, then they didn't come back even though several cycles of editing, changing @clean to @file and back again.

So after editing the sentinels out, then @clean files saved as expected.  But **something** must have caused those files to end up with orphaned sentinel lines.  After reproducing it for myself quite a few times over many days, now I can't intentionally cause the misbehavior.  This makes me think there is something lurking, waiting to get activated.  I don't know how to look for it now that I can't cause it to happen on purpose.

If I do get this behavior again I'll open a new issue with a more apt title.
